Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -1110,6 +1139,29 @@ checkCabalVersion pkg =
         ++ "that specifies the dependencies of the Setup.hs script itself. "
         ++ "The 'setup-depends' field uses the same syntax as 'build-depends', "
         ++ "so a simple example would be 'setup-depends: base, Cabal'."
+
+  , check (specVersion pkg >= Version [1,25] []
+           && elem (autogenModuleName pkg) (allModuleNames pkg)
```

Comment:
This made me double-take. It would be nice if we can rename `autogenModuleName` to `autogenPathsModuleName` (but keep the old function name around for BC; maybe deprecate it) so that it's clear that this is specifically about the `Paths` module.